### DISCUSSION TITLE
Started seeing System.ArgumentException (found BasicExpressionToken not MappingToken)

### DISCUSSION THREAD

**Discussion by gregschmit** [community/community#125059] [Actions] [answered]
### Select Topic Area

Bug

### Body

Today I started seeing this error on a repo that uses `snok/container-retention-policy`, however there hasn't been any recent changes to that action so it appears to be a bug in the action runners:

```
##[debug]action.yml for action: '/home/runner/work/_actions/snok/container-retention-policy/v2/action.yml'.
Error: System.ArgumentException: Unexpected type 'BasicExpressionToken' encountered while reading 'env'. The type 'MappingToken' was expected.
   at GitHub.DistributedTask.ObjectTemplating.Tokens.TemplateTokenExtensions.AssertMapping(TemplateToken value, String objectDescription)
   at GitHub.Runner.Worker.ActionManifestManager.ConvertRuns(IExecutionContext executionContext, TemplateContext templateContext, TemplateToken inputsToken, String fileRelativePath, MappingToken outputs)
   at GitHub.Runner.Worker.ActionManifestManager.Load(IExecutionContext executionContext, String manifestFile)
Error: Failed to load snok/container-retention-policy/v2/action.yaml
##[debug]System.ArgumentException: Failed to load snok/container-retention-policy/v2/action.yaml
```

The `env` it's referring to looks like this (https://github.com/snok/container-retention-policy/blob/main/action.yml):

```
...
          "$TOKEN_TYPE"
      env:
          ACCOUNT_TYPE: "${{ inputs.account-type }}"
          ORG_NAME: "${{ inputs.org-name }}"
...
```

Not sure if it's related to the YAML in that file using 4-spaces for the mapping indent after `env:`?

I also noticed that in my VSCode I started seeing my github workflow show red lines under parts of the `snok/container-retention-policy` step (again, on a repo that hasn't changed in the past few weeks; so it was valid before and today just started seeing these red underlines):

![image](https://github.com/community/community/assets/14255284/931ac00f-f6cc-4b15-994a-1d3db42db6a4)

Finally, here is an action that ran perfectly fine yesterday, and then today threw this error: https://github.com/gregschmit/rails-rest-framework/actions/runs/9167963220/job/25236841960. I just clicked "re-run this job".

**Comment by grandmas-favorite** [comment]
Hi @gregschmit!

This was likely caused by [this incident ](https://www.githubstatus.com/incidents/nj47vccwm2zj)that was reported this morning. The incident has since been updated, and Actions should be operating normally now.

**Reply by gregschmit** [reply]
I just re-ran that job and it's still failing for the same reason: https://github.com/gregschmit/rails-rest-framework/actions/runs/9167963220/job/25242482718

**Comment by ChristopherHX** [comment, marked as answer]
I believe the action has the syntax error here: https://github.com/snok/container-retention-policy/blame/3f4828a74358cd42680e02853e5202ebff46bc2c/action.yaml#L72

There are two wip commits in the v2 branch one of them has a 6hours ago timestamp. it's possible that the older breaking wip commit got pushed today to v2

There is no `: `, but a `=`

Maybe use `@b95617ac96d929f7a37025691e1ad89be732fb46` or a tag as version you trust to not contain wip commits

**Reply by gregschmit** [reply]
Good catch! I saw their latest release was Dec 8 2023 and assumed either releases would show up there in the sidebar or at least be merged to master.
